Makefile: should error out if bash isn't installed

Trying to build cockroach without `bash` results in a very cryptic error (see https://www.cockroachlabs.com/blog/critters-in-a-jar-running-cockroachdb-in-a-freebsd-jail/#comment-2771120260)

```
gostuff% gmake install
Makefile:243: .bootstrap: No such file or directory
gmake: -c: Command not found
Makefile:223: recipe for target '.git/hooks/post-merge' failed
gmake: *** [.git/hooks/post-merge] Error 127
```

We should error out with a more helpful message.

We should also call this out in the prerequisites in CONTRIBUTING.MD and in the build instructions (https://www.cockroachlabs.com/docs/install-cockroachdb.html).

--- a/Makefile
+++ b/Makefile
@@ -48,6 +48,9 @@ export PATH := $(GOPATH)/bin:$(PATH)
 # is one way to do this globally.
 # http://stackoverflow.com/questions/8941110/how-i-could-add-dir-to-path-in-makefile/13468229#13468229
 SHELL := $(shell which bash)
+ifeq ($(SHELL),)
+$(error bash is required)
+endif
 export GIT_PAGER :=
 
 # Note: We pass `-v` to `go build` and `go test -i` so that warnings

